File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        
        int left = 0;
        int right = numbers.size() - 1;
        
        while (true) {
            if (numbers[left] + numbers[right] < target) {
                left += 1;
            }
            else if (numbers[left] + numbers[right] > target) {
                right -= 1;
            }
            else {
                return {left+1, right+1};
            }
        }
        
    }
};

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        
        myDict = {}

        for idx, num in enumerate(numbers):
            if target-num in myDict:
                return [myDict[target-num] + 1, idx+1] 
            else:
                myDict[num] = idx
        return